src/components/lift-item/lift-item.tsx

```diff
@@ -0,0 +1,152 @@
+import { ExpandMore as ExpandMoreIcon } from '@material-ui/icons';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import React, { CSSProperties } from 'react';
+import { AntTabs, AntTab, TabPanel } from '../tab';
+import { LiftInformation } from './lift-item-information';
+import LiftRequestForm from './lift-item-form';
+import { LiftRequestManager } from '../../lift-state-manager';
+
+export interface LiftItemProps extends Omit<ExpansionPanelProps, 'children'> {
+  lift: Readonly<RomiCore.Lift>;
+  liftState?: Readonly<RomiCore.LiftState>;
+  enableRequest?: boolean;
+  onRequest?(
+    lift: RomiCore.Lift,
+    door_state: number,
+    request_type: number,
+    destination: string,
+  ): void;
+}
+
+export const LiftItem = React.forwardRef(function(
+  props: LiftItemProps,
+  ref: React.Ref<HTMLElement>,
+): React.ReactElement {
+  const { lift, liftState, enableRequest, onRequest, ...otherProps } = props;
+  const [tabValue, setTabValue] = React.useState(0);
+  const classes = useStyles();
+
+  function liftFloorLabel(liftState?: RomiCore.LiftState): string {
+    if (!liftState) {
+      return classes.liftFloorLabelUnknown;
+    }
+    switch (liftState.motion_state) {
+      case RomiCore.LiftState.MOTION_UP:
+      case RomiCore.LiftState.MOTION_DOWN:
+        return classes.liftFloorLabelMoving;
+      default:
+        return classes.liftFloorLabelStopped;
+    }
+  }
+
+  function handleRequest(doorState: number, requestType: number, destination: string): void {
+    !!onRequest && onRequest(lift, doorState, requestType, destination);
+  }
+
+  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
+    setTabValue(newValue);
+  };
+
+  return (
+    <React.Fragment>
+      <ExpansionPanel ref={ref} {...otherProps}>
+        <ExpansionPanelSummary
+          classes={{ content: classes.expansionSummaryContent }}
+          expandIcon={<ExpandMoreIcon />}
+        >
+          <Typography variant="h5">{lift.name}</Typography>
+          <Typography className={liftFloorLabel(liftState)} variant="button">
+            {liftState ? liftState.current_floor : 'Unknown'}
+          </Typography>
+        </ExpansionPanelSummary>
+        <ExpansionPanelDetails className={classes.expansionDetail}>
+          <AntTabs value={tabValue} onChange={handleChange} aria-label="ant example">
+            <AntTab label="Info" />
+            <AntTab label="Request" />
+          </AntTabs>
+          <TabPanel value={tabValue} index={0}>
+            <LiftInformation lift={lift} liftState={liftState} />
+          </TabPanel>
+          <TabPanel value={tabValue} index={1}>
+            {liftState?.available_floors && (
+              <LiftRequestForm
+                liftRequest={handleRequest}
+                doorStateList={LiftRequestManager.getListOfDoorModes()}
+                requestTypeList={LiftRequestManager.getListOfLiftRequestModes()}
+                destinationList={liftState?.available_floors}
```

Comment:
```suggestion
                destinationList={lift.levels}
```
Looks like the list of levels is available on the building map, we can use that so that we don't need to depend on the lift state being available.

Reply:
@koonpeng there may be scenarios where a lift only serves a select number of floors in the building. This information is captured in the LiftState msg.

Reply:
https://github.com/osrf/traffic_editor/blob/df9e58434e03010ace631234fff2fc0190bda5b2/building_map_msgs/msg/Lift.msg#L2 I see a `levels` field here, doesn't it include only the levels the lift can go to?

Reply:
Ah my bad for overlooking that. Yes that field will contain the list of levels the lift can access.